cmd/metadoc/metadoc.go

```diff
@@ -0,0 +1,334 @@
+// Copyright 2021 Fugue, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metadoc
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// Utility to modify metadata of rego files.
+//
+// You can load this using `RegoMetaFromPath` or `RegoMetaFromString`.
+//
+// Then, you read and/or modify the public fields of this structure.  When
+// you're done, you can turn it into a string again using `String()` and pass
+// it to OPA or write it to disk.
+type RegoMeta struct {
+	lines []string // Lines in the file
+
+	PackageName     string // Optional package name
+	packageNameLine int    // Original package name line, -1 if not present.
+
+	Imports         map[Import]struct{} // Imports (without aliases)
+	originalImports map[Import]struct{} // Helper to determine new imports
+
+	metadocStartLine int                    // Start of metadoc, -1 if not present
+	metadocEndLine   int                    // End of metadoc, -1 if not present
+	metadoc          map[string]interface{} // Dynamic metadoc
+
+	Id          string
+	Title       string
+	Description string
+	Severity    string
+	Controls    map[string][]string
+
+	ResourceType     string // Resource type
+	resourceTypeLine int
+
+	InputType     string // Input type
+	inputTypeLine int
+}
+
+func (meta *RegoMeta) GetFamilies() []string {
```

Comment:
I can help adapting this into the `metadoc` package, we probably want this as a read+write field as well so it can be updated from code.

Reply:
Sure.  This is the only code I added (I wanted to keep the diff minimal), but IMHO it would make sense to define some types and have Go parse the data automatically instead of just using `map[string]interface{}`.  If you want I can take a look at implementing that in Regula then tweak this PR to use it once its merged...

Reply:
Yeah, I think we can do that, it should be relatively low-effort to get this merged in Regula.